### PR TITLE
Support MacOs Arm target

### LINKED ISSUE #50: M1 Mac Support
Hi, I created a new project by following the wiki and when i ran the command `cargo make run`, I get the following errors:

`OpenGL ES 3.0 Renderer: Apple M1 Pro
Async. shader compilation: OFF

UNSUPPORTED (log once): POSSIBLE ISSUE: unit 1 GLD_TEXTURE_INDEX_2D is unloadable and bound to sampler type (Float) - using zero texture because texture unloadable
Registered camera FaceTime HD Camera with id 1 position 0 at index 0
ERROR: Can't open dynamic library: /Users/user/Developement/godot/godotshop/godot/../lib/x86_64-apple-darwin/libgodotshop.dylib, error: dlopen(/Users/user/Developement/godot/godotshop/godot/../lib/x86_64-apple-darwin/libgodotshop.dylib, 0x0002): tried: '/Users/user/Developement/godot/godotshop/godot/../lib/x86_64-apple-darwin/libgodotshop.dylib' (mach-o file, but is an incompatible architecture (have (x86_64), need (arm64e))), '/Users/user/Developement/godot/godotshop/lib/x86_64-apple-darwin/libgodotshop.dylib' (mach-o file, but is an incompatible architecture (have (x86_64), need (arm64e))).
   at: open_dynamic_library (platform/osx/os_osx.mm:1956)
ERROR: No valid library handle, can't get symbol from GDNative object
   at: get_symbol (modules/gdnative/gdnative.cpp:510)
ERROR: No nativescript_init in "res://../lib/x86_64-apple-darwin/libgodotshop.dylib" found
   at: init_library (modules/gdnative/nativescript/nativescript.cpp:1503)`

Is M1 Mac not supported with this template? Thank you

### PR DESCRIPTION
**Pull request by finnhartshorn**
Adds support for arm on MacOs. `aarch64-apple-darwin` is used by default on MacOs arm machines and I've added in commented out values to switch to x86_64 on Arm or vice versa.

Closes #50

### CODE REVIEW

File: Makefile.toml [hunk 1]
```diff
@@ -222,9 +224,15 @@ dependencies = ["build-x86_64-unknown-linux-gnu-debug"]
 run_task = "run-godot"
 
 [tasks.run-for-macos]
+# dependencies = ["build-aarch64-apple-darwin-debug"]
```
Comment (macalimlim) [inline Makefile.toml:227]: Can be removed since both macOS tasks are separated.

File: Makefile.toml [hunk 2]
```diff
@@ -222,9 +224,15 @@ dependencies = ["build-x86_64-unknown-linux-gnu-debug"]
...
 dependencies = ["build-x86_64-apple-darwin-debug"]
 run_task = "run-godot-macos"
 
+[tasks.run-for-macos-apple-silicon]
+dependencies = ["build-aarch64-apple-darwin-debug"]
+# dependencies = ["build-x86_64-apple-darwin-debug"]
```
Comment (macalimlim) [inline Makefile.toml:233]: Same as above, can be removed also

**Review by macalimlim — CHANGES_REQUESTED** [review]
Also, please squash your commits into one commit

File: Makefile.toml (hunk 1 above)
Comment (finnhartshorn) [reply Makefile.toml]: Done

File: Makefile.toml (hunk 2 above)
Comment (finnhartshorn) [reply Makefile.toml]: And done

**Comment by macalimlim** [comment]
Thank you for this PR and great job! 👍🏾

**Pull request merged**